build/tsc.gni

```diff
@@ -0,0 +1,40 @@
+import("npm.gni")
+
+template("typescript_build") {
+  assert(defined(invoker.tsconfig),
+         "Need tsconfig name to run (must exist in electron root)")
+  assert(defined(invoker.sources), "Need tsc sources to run")
+  assert(defined(invoker.electron_dir),
+         "Need electron_dir to run, should be 'lib' or other top level dir")
+
+  npm_action(target_name) {
+    forward_variables_from(invoker,
+                           [
+                             "deps",
+                             "public_deps",
+                             "outputs",
+                           ])
+    script = "tsc"
+
+    sources = invoker.sources
+
+    args = [
+      "-p",
+      "tsconfig." + invoker.tsconfig + ".json",
```

Comment:
- the tsconfig needs to be listed in the inputs
- don't construct the name of the file by string concatenation, just specify the name of the file in the variable. i.e. `tsconfig = "tsconfig.electron.json"`. resolve it here with `rebase_path`.